src/Domain.UnitTests/EarningsProfileGenerator_GenerateEarningsTests.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using NServiceBus;
+using NUnit.Framework;
+using SFA.DAS.Apprenticeships.Types;
+using SFA.DAS.Funding.ApprenticeshipEarnings.Types;
+
+namespace SFA.DAS.Funding.ApprenticeshipEarnings.Domain.UnitTests;
+
+public class EarningsProfileGenerator_GenerateEarningsTests
+{
+    private EarningsProfileGenerator _sut;
+    private ApprenticeshipCreatedEvent _apprenticeshipLearnerEvent;
+    private Mock<IOnProgramTotalPriceCalculator> _mockOnProgramTotalPriceCalculator;
+    private Mock<IInstallmentsGenerator> _mockInstallmentsGenerator;
+    private Mock<IMessageSession> _mockMessageSession;
+    private Mock<IEarningsGeneratedEventBuilder> _mockEarningsGeneratedEventBuilder;
+    private decimal _expectedAdjustedPrice;
+    private List<EarningsInstallment> _expectedEarningsInstallments;
+    private EarningsGeneratedEvent _expectedEarningsGeneratedEvent;
+    private EarningsProfile _result;
+
+    [SetUp]
+    public async Task SetUp()
+    {
+        _apprenticeshipLearnerEvent = new ApprenticeshipCreatedEvent
+        {
+            FundingType = FundingType.NonLevy,
+            ActualStartDate = new DateTime(2022, 8, 1),
+            ApprenticeshipKey = Guid.NewGuid(),
+            EmployerAccountId = 114,
+            PlannedEndDate = new DateTime(2024, 7, 31),
+            UKPRN = 116,
+            TrainingCode = "able-seafarer",
+            FundingEmployerAccountId = 118,
+            Uln = "900000118",
+            AgreedPrice = 15000,
+            ApprovalsApprenticeshipId = 120,
+            LegalEntityName = "MyTrawler"
```

Comment:
be good to use AutoFixture here for more randomised inputs

Reply:
done